### ISSUE TITLE
Plugin loading and binary incompatibility - part 2

### ISSUE THREAD

**Issue by mhier** [ChimeraTK/DeviceAccess#28]
This issue depends on #27. Please read that ticket first for the background information.

The idea is to address the first problem:
- The version of DeviceAccess used to compile the plugin is incompatible with the one used to compile the application

This problem can only be solved by somehow loading the right version of the plugin, assuming it has been installed on the system. For this, the general loading mechanism needs to be changed.

I see two alternative solutions for this problem:
- Use a central registry which maps the DeviceAccess version on to the right so file (for each plugin name), or
- enumerate all plugin so files matching a given pattern and find the version which is compiled against the right DeviceAccess version.

Before working on this ticket, the decision which alternative to implement must be made.

The current loading mechanism with @LOAD_LIB probably should stay unaltered but should be marked as deprecated (at least if it is fully superseded by the new mechanism).

**Comment by mhier** [comment]
New tickets will be created for this issue.
